flink-table/flink-table-planner/src/main/java/org/apache/calcite/avatica/util/DateTimeUtils.java

```diff
@@ -783,6 +840,30 @@ private static long firstMondayOfFirstWeek(int year) {
 		return janFirst + (11 - janFirstDow) % 7 - 3;
 	}
 
+	/** Returns the ISO-8601 week number based on year, month, day.
+	 * Per ISO-8601 it is the Monday of the week that contains Jan 4,
+	 * or equivalently, it is a Monday between Dec 29 and Jan 4.
+	 * Sometimes it is in the year before the given year, sometimes after. */
+	private static int getIso8601WeekNumber(int julian, int year, int month, int day) {
+		long fmofw = firstMondayOfFirstWeek(year);
+		if (month == 12 && day > 28) {
+			if (31 - day + 4 > 7 - ((int) floorMod(julian, 7) + 1)
```

Comment:
Although the Calcite bug may be fixed in future, lets's add UT for these transformation functions for our current code base.

Reply:
these lines are copied from avatica 1.14.0 release